app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -1148,16 +1148,27 @@ protected function _exportCsvItem(Varien_Object $item, Varien_Io_File $adapter)
     /**
      * Retrieve a file container array by grid data as CSV
      *
-     * Return array with keys type and value
+     * Return array with keys type and value [the original way]
+     * Return array with keys file name, type and value, file mime type [the new way]
      *
+     *  the original way (still working): $this->_prepareDownloadResponse($fileName, $grid->getCsvFile($fileName));
+     *      or (with right content type): $this->_prepareDownloadResponse($fileName, $grid->getCsvFile($fileName), 'text/csv');
+     *                       the new way: $this->_prepareDownloadResponse(...$grid->getCsvFile($fileName, -1));
+     *
+     * @param string $fileName
+     * @param int $limit
      * @return array
      * @throws Exception
      */
-    public function getCsvFile()
+    public function getCsvFile($fileName = '', $limit = 0)
```

Comment:
when we modify a public method it's always BC, somebody could have extended it and, adding parameters, the method declaration would no longer be compliant with the father class.

Reply:
hum, yes :)